datasource-local/src/main/java/cmm/apps/esmorga/datasource_local/database/dao/EventDao.kt

```diff
@@ -11,14 +11,17 @@ import cmm.apps.esmorga.datasource_local.event.model.EventLocalModel
 interface EventDao {
 
     @Insert(onConflict = OnConflictStrategy.REPLACE)
-    suspend fun insertEvent(event: List<EventLocalModel>)
+    suspend fun insertEvents(event: List<EventLocalModel>)
 
     @Query("SELECT * FROM EventLocalModel")
     suspend fun getEvents(): List<EventLocalModel>
 
     @Query("DELETE FROM EventLocalModel")
     suspend fun deleteAll()
 
-    @Query( "SELECT * FROM EventLocalModel WHERE localId = :eventId")
+    @Query("SELECT * FROM EventLocalModel WHERE localId = :eventId")
     suspend fun getEventById(eventId: String): EventLocalModel
+
+    @Query("UPDATE EventLocalModel SET localUserJoined = :userJoined WHERE localId = :eventId")
+    suspend fun updateEventJoinedById(eventId: String, userJoined: Boolean)
```

Comment:
Maybe it's better to create a "updateEvent" receiving the localModel and with @Insert(onConflic = REPLACE), then we can reuse the method for any local change that we could need.

Also you are calling updateEvenJoined but then you are calling it in "leaveEvent" which is not the better naming. 

Creating an updateEvent function will solve both scenarios.